Contributing.md

```diff
@@ -1,30 +1,472 @@
+  place. The status RFC will then change to ![WIP status](./RFC/src/theme/images/status-wip.svg). A feature flag 
+  name will be assigned to the feature at this stage.
+- Development for this feature happens behind the feature flag, and PRs are submitted against the `development` branch.
+- Once the implementation is complete and active on mainnet, the RFC status can then be changed to
+  ![active status](./RFC/src/theme/images/status-active.svg).
+- The RFC should be updated to match the implementation. The changes must be summarised in the Change Log. Once the 
+  feature is stabilised and is live on mainnet, the status can be updated to
+  ![stable status](./RFC/src/theme/images/status-stable.svg).
+
+If an RFC ever becomes redundant, DO NOT delete it. Mark the RFC as 
+![deprecated](./RFC/src/theme/images/status-deprecated.svg) and move it to the _Deprecated_ chapter of the RFC handbook.
+
+
+The source code must ultimately be the source of truth for the Tari implementation.
+If the code and RFC have deviated substantially, mark it ![out of date](./RFC/src/theme/images/status-outofdate.svg).
+Then file an issue asking someone to kindly update the RFC and bring it back in line with the code.
+
+Taking on this thankless task is an excellent way for new contributors to learn the code base and quickly add value 
+to the project!
+
+[Discord]: https://discord.gg/q3Sfzb8S2V
+
+## Pull requests
+
+You're submitting your first, or hundredth, PR to the Tari codebase. Congratulations! The core team could really use 
+the help. But they also need to be super careful since after mainnet launch, the Tari code is managing real money. 
+
+It's therefore typical and even expected for Pull Requests to undergo several revisions before being merged. You 
+can expect (constructive) feedback and suggestions for improvements that will typically reflect the ideas espoused 
+in the following set of guidelines.
+
+### PRs do one job
+[do one job]: #prs-do-one-job
+
+This is really important. A single PR needs to address a single thing. This is not Congress, where bills have 
+dozens of unrelated things wadded into them in the hope that they're more likely to sneak through with the main 
+proposal.
+
+Keeping PRs focused on one thing has many benefits:
+- It keeps PRs [small](#prs-are-small).
+- It maintains focus. It would suck if an entire PR was blocked from being merged because some other unrelated 
+  change was causing an issue.
+
+If a PR tries too much multi-tasking, it will likely be labelled `CR-one_job` and parked. The solution is 
+simple: Break the PR up into 2 or more PRs, each addressing a single issue. Mention in the git comments that you 
+have done this.
+
+
+### PRs are small
+
+PRs should be under 400 lines long. 
+
+Unit tests do not count towards this line count.
+
+Documentation and RFCs do not contribute to line count. However, RFC PRs should generally not have any code in the 
+at all (since PRs [do one job]). 
```

Comment:
```suggestion
Documentation and RFCs do not contribute to line count. However, RFC PRs should generally not have any code in it 
at all (since PRs [do one job]). 
```